server_metrics.go

```diff
@@ -4,29 +4,91 @@ import (
 	"context"
 	"github.com/grpc-ecosystem/go-grpc-prometheus/packages/grpcstatus"
 	prom "github.com/prometheus/client_golang/prometheus"
+	"sync"
 
 	"google.golang.org/grpc"
 )
 
+var (
+	lock                    sync.RWMutex
+	defaultServerMetrics    *ServerMetrics
+	unaryServerInterceptor  grpc.UnaryServerInterceptor
+	streamServerInterceptor grpc.StreamServerInterceptor
```

Comment:
having global variable doesn't seem like a good idea.

Reply:
True, it is by initial design. See `server.go`.